pkg/asset/installconfig/azure/client.go

```diff
@@ -211,3 +212,60 @@ func (c *Client) ListResourceIDsByGroup(ctx context.Context, groupName string) (
 	}
 	return res, nil
 }
+
+// GetVirtualMachineSku retrieves the restource SKU of a specified virtual machine in the given region.
+func (c *Client) GetVirtualMachineSku(ctx context.Context, name, region string) (azsku.ResourceSku, error) {
+	results, err := c.GetResourceSkus(ctx, "virtualMachines", region)
+	if err != nil {
+		return azsku.ResourceSku{}, err
+	}
+
+	for _, sku := range results {
+		if strings.EqualFold(name, *sku.Name) {
+			return sku, nil
+		}
+	}
+
+	return azsku.ResourceSku{}, errors.Errorf("SKU not found for specified subscription in region %s", region)
+
+}
+
+// GetResourceSkus retrieves the resource SKUs of a specified type for a given region.
+func (c *Client) GetResourceSkus(ctx context.Context, resourceType, region string) ([]azsku.ResourceSku, error) {
```

Comment:
I don't think that we have any need for this function outside of getting a specific SKU, so let's forego returning **all** of the SKUs in the region. Returning all of the SKUs in a slice exposes us to crashes in the (unlikely) event that the server responds with more SKUs than we can handle.

Reply:
Fixed in https://github.com/openshift/installer/commit/91402a22c47edd67c4b9802c4f2e47bfce3b3f39